src/modules/85-iacm/components/IACMPipelineSteps/IACMApproval/__tests__/index.test.tsx

```diff
@@ -0,0 +1,115 @@
+import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
+import { factory, TestStepWidget } from '@pipeline/components/PipelineSteps/Steps/__tests__/StepTestUtil'
+
+import { IACMApprovalStep } from '..'
+
+describe('IACMApprovalStep Interaction', () => {
+  beforeEach(() => {
+    factory.registerStep(new IACMApprovalStep())
+  })
+
+  test('renders form', () => {
+    const initialValues = {
+      type: StepType.IACMApproval,
+      name: 'IACMApproval',
+      identifier: 'IACMApproval',
+      timeout: '10m'
+    }
+
+    const { getByText } = render(
+      <TestStepWidget initialValues={initialValues} type={StepType.IACMApproval} stepViewType={StepViewType.Edit} />
+    )
+
+    const nameField = getByText('Name')
+    const timeoutField = getByText('pipelineSteps.timeoutLabel')
+
+    expect(nameField).toBeInTheDocument()
+    expect(timeoutField).toBeInTheDocument()
+  })
+
+  test('submits valid form', async () => {
+    const ref = createRef<StepFormikRef<unknown>>()
+    const onUpdate = jest.fn()
+    const initialValues = {
+      type: StepType.IACMApproval,
+      name: 'IACMApproval',
+      timeout: '1h',
+      identifier: 'IACMApproval'
+    }
+
+    render(
+      <TestStepWidget
+        initialValues={initialValues}
+        type={StepType.IACMApproval}
+        stepViewType={StepViewType.Edit}
+        onUpdate={onUpdate}
+        ref={ref}
+      />
+    )
+
+    await act(() => ref.current?.submitForm()!)
+
+    expect(onUpdate).toHaveBeenCalled()
+  })
+
+  test('errors when submitting invalid form', async () => {
+    const ref = createRef<StepFormikRef<unknown>>()
+    const onUpdate = jest.fn()
+    const initialValues = {
+      type: StepType.IACMApproval,
+      name: 'IACMApproval',
+      timeout: '10h',
+      identifier: 'IACMApproval'
+    }
+
+    const { getByText } = render(
+      <TestStepWidget
+        initialValues={initialValues}
+        type={StepType.IACMApproval}
+        stepViewType={StepViewType.Edit}
+        onUpdate={onUpdate}
+        ref={ref}
+      />
+    )
+
+    await act(() => ref.current?.submitForm()!)
+
+    const error = getByText('During IACM Beta, timeout value must be less than or equal to 1h')
+    expect(error).toBeInTheDocument()
+    expect(onUpdate).not.toHaveBeenCalled()
+  })
+})
+
+describe('IACMApprovalStep Validation', () => {
+  test('minimum timeout value', () => {
+    const response = new IACMApprovalStep().validateInputSet({
+      data: { name: 'IACMApprovalValidation', identifier: 'IACMApprovalValidation', timeout: '1s' },
+      viewType: StepViewType.TriggerForm,
+      template: {
+        name: 'IACMApprovalValidation',
+        identifier: 'IACMApprovalValidation',
+        timeout: '<+input>'
+      }
+    })
+
+    expect(response.timeout).toBe('Value must be greater than or equal to "10s"')
+  })
+
+  test('maximum timeout value', () => {
```

Comment:
Ideally test names should describe what you expect to happen
```suggestion
  test('it should display an error when the timeout value is more than 1h', () => {
```

That way if it fails later, you don't start with `IACMApprovalStep Validation - maximum timeout value`, instead you immediately see what's failing without having to first try to figure out what the test is doing.

Reply:
Fair point! Although `should` doesn't belong in test names - it either does or it does not lol

Reply:
I like to ask my tests politely to pass rather than demanding. They can be childish some times. If you want to substitute `will` in its place, fire away.